src/Markdig/Syntax/MarkdownDocument.cs

```diff
@@ -15,5 +15,10 @@ public class MarkdownDocument : ContainerBlock
         public MarkdownDocument() : base(null)
         {
         }
+
+        /// <summary>
+        /// Gets the number of lines in this <see cref="MarkdownDocument"/>
+        /// </summary>
+        public int Lines { get; internal set; }
```

Comment:
Better use `LineCount` instead (Lines can imply that it would return an array). You can also leave it public get set (not internal). It is not something important to protect and doesn't hurt if someone wants to change it (create a MarkdownDocument manually...etc.)

Reply:
Commit amended.
As to why, I wrote a tool that checks a bunch of things about markdown 'codebases', and emits warnings. I wrote [my own regex-y parser](https://github.com/MihaZupan/MarkdownValidator/blob/master/src/ReferenceValidator/Parser/MarkdownFile.cs#L88) to detect the structures I needed, but now I decided it would be much easier and much more reliable if I just use a well-established parser.

On another note, I was going to accept a pipeline builder, but now I realise that it's not designed for over multiple files. Should I make users implement some sort of class that returns fresh instances of pipeline/pipeline builder every time? Or is there a cleaner solution I'm missing here?